core/embed/trezorhal/stm32f4/flash.c

```diff
@@ -167,6 +167,50 @@ secbool flash_area_erase_bulk(const flash_area_t *area, int count,
   return sectrue;
 }
 
+secbool flash_area_erase_partial(const flash_area_t *area, uint32_t offset,
+                                 uint32_t *_bytes_erased) {
+  uint32_t sector_offset = 0;
+  *_bytes_erased = 0;
+
+  for (int s = 0; s < area->num_subareas; s++) {
+    for (int i = 0; i < area->subarea[s].num_sectors; i++) {
+      uint32_t sector_index = area->subarea[s].first_sector + i;
+      uint32_t sector_size = FLASH_SECTOR_TABLE[sector_index + 1] -
+                             FLASH_SECTOR_TABLE[sector_index];
+
+      if (offset == sector_offset) {
```

Comment:
what is the reason not to accept any offset within the sector? its not necessarily wrong but its also not super-clear from the function parameter names nor the comment, to me at least.

Reply:
I slightly updated the comment to make it more clear.  It's not possible to erase just part of the sector, so function fails if the caller is not precise in specifying what to erase.  see fd93c86278676cf67b186afe7bae15548652a462